src/search/flowr-search-filters.ts

```diff
@@ -3,29 +3,47 @@ import type { VertexType } from '../dataflow/graph/vertex';
 import { ValidVertexTypes } from '../dataflow/graph/vertex';
 import type { NormalizedAst, ParentInformation } from '../r-bridge/lang-4.x/ast/model/processing/decorate';
 import type { DataflowInformation } from '../dataflow/info';
-import type { RNode } from '../r-bridge/lang-4.x/ast/model/model';
+import type { FlowrSearchElement } from './flowr-search';
+import type { MergeableRecord } from '../util/objects';
+import type { Enrichment } from './search-executor/search-enrichers';
+import { enrichmentContent } from './search-executor/search-enrichers';
 
 export type FlowrFilterName = keyof typeof FlowrFilters;
+interface FlowrFilterWithArgs<Filter extends FlowrFilterName, Args extends FlowrFilterArgs<Filter>> {
+	name: Filter;
+	args: Args
+}
 
 export enum FlowrFilter {
-	DropEmptyArguments = 'drop-empty-arguments'
+	DropEmptyArguments = 'drop-empty-arguments',
```

Comment:
On that note, we should probably add at least tsdoc comments for what these mean, whether they take arguments etc. and add them to the documentation.